oci/layout/oci_dest.go

```diff
@@ -224,3 +230,30 @@ func (d *ociImageDestination) Commit() error {
 	}
 	return ioutil.WriteFile(d.ref.indexPath(), indexJSON, 0644)
 }
+
+func ensureDirectoryExists(path string) error {
+	if _, err := os.Stat(path); err != nil && os.IsNotExist(err) {
+		if err := os.MkdirAll(path, 0755); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ensureParentDirectoryExists ensures the parent of the supplied path exists.
+func ensureParentDirectoryExists(path string) error {
+	return ensureDirectoryExists(filepath.Dir(path))
+}
+
+// Exists checks whether the specified path exists.
+// The implementation is opinionated, since in case of unexpected errors false is returned
+func exists(path string) bool {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true
+	}
+	if os.IsNotExist(err) {
+		return false
+	}
+	return true
```

Comment:
This can be implemented as

```
func exists(path string) bool {
    _, err := os.Stat(path)
    return err != nil && os.IsNotExist(err)
}
```

I also think that you could even remove the `err != nil` check (`os.IsNotExist(nil) == false`).